### DISCUSSION TITLE
wss non-blocking read with libcr

### DISCUSSION THREAD

**Discussion by pmuvb** [curl/curl#15551] [General]
Hi,

I know it's work in progress, but are you planning to have a blocking read in wss? Or, maybe you planning to provide a 
"poll" function?

Currently it's non-blocking.

I've started using it as a client, there are two things happen:
1. it reads incomplete strings from wss, not every time and all the time but quite often to make a problem for me since those are JSON strings.

2. it spins the CPU at 100% and heats up my laptop.

Thanks!

EDIT: I am using clang++ v.18 on Debian 12, libcurl 8.11.0

**Comment by pmuvb** [comment]
Not a single answer.

I believe this is a legitimate question - are there plans for a blocking read?

Thanks

**Comment by dfandrich** [comment]
The man page `curl_ws_recv()` says *Instead of blocking, the function returns CURLE_AGAIN. The correct behavior is then to wait for the socket to signal readability before calling this function again.*  Is the code doing that?

**Comment by pmuvb** [comment]
I am just ignoring error 81 and go to the next iteration to call curl_ws_recv().

Here is the code excerpt: https://pastebin.com/fsve9HRh

How do I "wait for the socket to signal readability" ?

thanks

**Comment by pmuvb** [comment]
FYI: I've moved on and running on a different library till you guys sort it out your websocket.
